src/build.tasks

```diff
@@ -0,0 +1,43 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="12.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <UsingTask TaskName="PInvokesExports" TaskFactory="CodeTaskFactory" AssemblyFile="$(MSBuildToolsPath)\Microsoft.Build.Tasks.v12.0.dll">
+    <ParameterGroup>
+      <LibName ParameterType="System.String" />
+      <ExportPath ParameterType="System.String" />
+      <AssemblyPath ParameterType="System.String" />
+    </ParameterGroup>
+    <Task>
+      <Using Namespace="System.IO" />
+      <Using Namespace="System.Text" />
+      <Using Namespace="System.Reflection" />
+      <Using Namespace="System.Runtime.InteropServices" />
+      <Code Type="Fragment" Language="cs">
+        <![CDATA[          
+            StringBuilder sb = new StringBuilder();
+            string assemblyPath = string.Format(@"{0}PInvoke.{1}.dll", AssemblyPath, LibName);
```

Comment:
Instead of trying to recreate the path to the assembly, why not pass `$(TargetPath)` into the Task from the MSBuild targets file?